src/compiler/checker.ts

```diff
@@ -18293,6 +18293,13 @@ namespace ts {
                 error(indexExpression, Diagnostics.A_const_enum_member_can_only_be_accessed_using_a_string_literal);
                 return errorType;
             }
+            if (isTupleType(objectType) && !objectType.target.hasRestElement && isNumericLiteral(indexExpression)) {
+                const index = +indexExpression.text;
+                const maximumIndex = length(objectType.target.typeParameters);

```

Comment:
Looks like there should be `length(...) - 1` to get last available index. Otherwise errors looks off by 1 and confusing.

P.S. Thank you for PR. Glad to see that this will be in TS soon. 👍

Reply:
sorry, my fault